src/edu/csus/ecs/pc2/core/StringUtilities.java

```diff
@@ -386,7 +386,41 @@ public static Integer getTeamNumber(String userLogin) {
     public static String removeLastChar(String s) {
         return (s == null || s.length() == 0) ? null : (s.substring(0, s.length()-1));
     }
-    
 
+    public static String removeUpTo(String source, String stringToRemove) {
```

Comment:
This method should have JavaDoc explaining what it does, especially since the name `removeUpTo` is not especially clear.  "Up to" what?  The second parameter is named "stringToRemove"; does it remove "up to" that string?  Does removal include removing the specified string?  What happens if the specified string isn't found in the source string?  What happens if the specified string occurs more than once in the source string?

(Yes, all of these things can be figured out by looking at the code -- but that's technically true for _any_ method; header comments are precisely for the purpose of explaining what a method does _without having to examine the code_.)